core/src/main/java/org/elasticsearch/discovery/zen/PublishClusterStateStats.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.discovery.zen;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Class encapsulating stats about the PublishClusterStateAction
+ */
+public class PublishClusterStateStats implements Writeable, ToXContentFragment {
+
+    private final long fullClusterStateSentCount;
+    private final long clusterStateDiffSentCount;
+    private final long incompatibleClusterStateDiffVersionCount;
+
+    public PublishClusterStateStats(long fullClusterStateSentCount,
+                                    long clusterStateDiffSentCount,
+                                    long incompatibleClusterStateDiffVersionCount) {
+        this.fullClusterStateSentCount = fullClusterStateSentCount;
+        this.clusterStateDiffSentCount = clusterStateDiffSentCount;
+        this.incompatibleClusterStateDiffVersionCount = incompatibleClusterStateDiffVersionCount;
+    }
+
+    public PublishClusterStateStats(StreamInput streamInput) throws IOException {
```

Comment:
We normally name these `in`, it leaves for shorter code and it would be consistent with other places in the codebase.